server/src/test/java/org/elasticsearch/script/JodaCompatibleZonedDateTimeTests.java

```diff
@@ -0,0 +1,211 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.elasticsearch.common.logging.Loggers;
+import org.elasticsearch.test.ESTestCase;
+
+import java.security.AccessControlContext;
+import java.security.AccessController;
+import java.security.PermissionCollection;
+import java.security.Permissions;
+import java.security.PrivilegedAction;
+import java.security.ProtectionDomain;
+import java.time.DayOfWeek;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.ZoneOffset;
+import java.util.Locale;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class JodaCompatibleZonedDateTimeTests extends ESTestCase {
+    private static final Logger DEPRECATION_LOGGER =
+        LogManager.getLogger("org.elasticsearch.deprecation.script.JodaCompatibleZonedDateTime");
+
+    // each call to get or getValue will be run with limited permissions, just as they are in scripts
+    private static PermissionCollection NO_PERMISSIONS = new Permissions();
+    private static AccessControlContext NO_PERMISSIONS_ACC = new AccessControlContext(
+        new ProtectionDomain[] {
+            new ProtectionDomain(null, NO_PERMISSIONS)
+        }
+    );
+
+    // Thursday, September 6, 2018 3:01:20.617 PM GMT-07:00 DST
+    private static JodaCompatibleZonedDateTime TIME =
+        new JodaCompatibleZonedDateTime(Instant.ofEpochMilli(1536271280617L), ZoneOffset.ofHours(-7));
+
+    void assertDeprecation(Runnable assertions, String message) {
+        Appender appender = new AbstractAppender("test", null, null) {
+            @Override
+            public void append(LogEvent event) {
+                /* Create a temporary directory to prove we are running with the
+                 * server's permissions. */
+                createTempDir();
+            }
+        };
+        appender.start();
+        Loggers.addAppender(DEPRECATION_LOGGER, appender);
+        try {
```

Comment:
Maybe a comment here like "now run the block with the same permissions as a script".